site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -25,49 +28,78 @@
 @Service
 public class SiteServiceImpl implements SiteService {
 
+    private final Map<String, SiteConfiguration> sitesByType = new HashMap<>();
+    private final Map<String, String> parentTypeByChildType = new HashMap<>();
     private final EntityStore<Site, String> siteStore;
     private final ModelEntityValidation<Site> entityValidation;
     private final Logger logger = LoggerFactory.getLogger(SiteServiceImpl.class);
 
     /**
+     * @param configurationProvider provide Site Configuration
      * @param siteStore        Site store interface
      * @param entityValidation Site validation interface
      */
     @Autowired
-    public SiteServiceImpl(final EntityStore<Site, String> siteStore,
+    public SiteServiceImpl(SiteConfigurationProvider configurationProvider,
+                           final EntityStore<Site, String> siteStore,
                            ModelEntityValidation<Site> entityValidation) {
+        if (configurationProvider == null) {
+            throw new InitialisationError("site configuration provider cannot be null");
+        }
         if (siteStore == null) {
             throw new InitialisationError("site store cannot be null");
         }
         if (entityValidation == null) {
             throw new InitialisationError("entity validation cannot be null");
         }
+        var configuration = configurationProvider.getConfiguration();
         this.siteStore = siteStore;
         this.entityValidation = entityValidation;
+        addTypesToMap(configuration);
         logger.info(Services.STARTUP.message());
     }
 
+    private void addTypesToMap(final SiteConfiguration configuration) {
+        this.sitesByType.put(configuration.getType(), configuration);
+        if (configuration.getChild() != null) {
+            for (final var childConfig : configuration.getChild()) {
+                addTypesToMap(childConfig);
+                this.parentTypeByChildType.put(childConfig.getType(), configuration.getType());
+            }
+        }
+    }
+
     /**
      * @param site the Site to save.
      * @return The id of the new site
      */
     @Override
     public String save(final Site site) {
         var validationResponse = entityValidation.validate(site);
```

Comment:
couldnt we just call a validate method and teh validate method throws the exceptions? I think the 'if' checks should be in the validator?

Reply:
We agreed that the validator should validate just the entity by itself, and the service does the checks on parent-child relationships etc. - that way the validator doesn't depend on the entity store.

Reply:
Agree with Alex